### PR TITLE
Added support to override auth_type in auth_file in pgbouncer helm configuration

### LINKED ISSUE #19654: Pgbouncer auth type cannot be configured
### Official Helm Chart version

1.3.0 (latest released)

### Apache Airflow version

2.2.1

### Kubernetes Version

1.20.9

### Helm Chart configuration

_No response_

### Docker Image customisations

_No response_

### What happened

The default pgbouncer config generated by the helm chart looks something like

```
[databases]
...

[pgbouncer]
pool_mode = transaction
listen_port = 6543
listen_addr = *
auth_type = md5
auth_file = /etc/pgbouncer/users.txt
stats_users = <user>
ignore_startup_parameters = extra_float_digits
max_client_conn = 200
verbose = 0
log_disconnections = 0
log_connections = 0

server_tls_sslmode = require
server_tls_ciphers = normal
```

If the database to connect against is Azure Postgresl, the auth type `md5` [does not seem](https://github.com/pgbouncer/pgbouncer/issues/325) to be supported. Hence, we need to add a configuration flag to change the auth type to something else in the helm chart. 

### What you expected to happen

_No response_

### How to reproduce

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by subkanthi**
Added support to override auth_type in auth_file in pgbouncer helm configuration

closes: #19654

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
LGTM. But I think @jedcunningham and @ephraimbuddy shoudl respond

File: chart/values.schema.json [hunk 1]
```diff
@@ -3327,6 +3327,16 @@
                     "type": "integer",
                     "default": 0
                 },
+                "auth_type": {
+                    "description": "Method of Authenticating users",
```
Comment (jedcunningham) [inline chart/values.schema.json]:
```suggestion
                    "description": "Method of authenticating users",
```

nit

**Review by jedcunningham — APPROVED** [review]

**Comment by subkanthi** [comment]
Resolved conflicts.

**Comment by subkanthi** [comment]
Can this one be merged. Thanks

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Merged @subkanthi :)

**Pull request merged**